### PR TITLE
FIX Badge component test: convert to Component and add truthy test

### PR DESCRIPTION
**Pull request by robbieaverill**
The original "should return null if status is empty" test was asserting that the component render result was null, but the component wasn't being executed at all, so the result was null by default.

### CODE REVIEW

File: client/src/components/Badge/tests/Badge-test.js [hunk 1]
```diff
@@ -1,32 +1,39 @@
...
+          className="customclass"
+        />
       );
 
-      expect(badge).toBe(null);
+      expect(badge.render()).toEqual(
+        <span className="customclass badge badge-success badge-pill">
+          Hello world
+        </span>
+      );
```
Comment (flamerohr) [inline client/src/components/Badge/tests/Badge-test.js]: I prefer cherry picking parts that you truly expect, than a whole dump of a big object...
so test that `badge-success` class exists
test that `Hello world` is in the `textContent`

That way unit tests don't ALWAYS break when we tweak the component

Comment (robbieaverill) [reply client/src/components/Badge/tests/Badge-test.js]: Sure. I’d argue the opposite in that changes to a component should be reflected in the tests too, but I agree that partial assertions would be totally fine

Comment (robbieaverill) [reply client/src/components/Badge/tests/Badge-test.js]: Updated =)

Comment (flamerohr) [reply client/src/components/Badge/tests/Badge-test.js]: Yeah, I'd argue with breaking that test down even more `it()` statements :D
That way you test each feature atomically, this refactor is good to go though.

Comment (robbieaverill) [reply client/src/components/Badge/tests/Badge-test.js]: Good counter! Thanks :-)

**Pull request merged**